sdk/src/beta9/runner/function.py

```diff
@@ -183,30 +189,24 @@ def main(channel: Channel):
 
         task_duration = time.time() - start_time
 
-        # End the task
-        end_task_response = gateway_stub.end_task(
-            EndTaskRequest(
+        end_task_response = end_task_and_send_callback(
```

Comment:
Have we tested when a task times out or is cancelled now that this we send the callback before monitor_task.cancel() is called?

Reply:
wdym? monitor tasks has its own send_callback functions

Reply:
task cancel and timeout and cancelled still works properly

Reply:
Does this invoke the callback just once if it times out?

Reply:
should invoke it just once.